.github/workflows/release.yaml

```
@@ -35,175 +36,104 @@ jobs:
-      #   uses: docker/setup-qemu-action@v1
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
 
-      - name: Generate image metadata
-        id: metadata
-        uses: docker/metadata-action@v4
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
         with:
-          images: prefecthq/prefect
-          tags: |
-            type=pep440,pattern={{version}},suffix=-python${{ matrix.python-version }}
-            type=pep440,pattern={{major}}.{{minor}},suffix=-python${{ matrix.python-version }}
-            type=pep440,pattern={{major}},suffix=-python${{ matrix.python-version }}
-            type=sha,suffix=-python${{ matrix.python-version }}
-            type=raw,value=2-latest,enable=${{ matrix.python-version == '3.10'}}
-          flavor: |
-            latest=false
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
 
       - name: Checkout
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
-      - name: Build image
-        uses: docker/build-push-action@v3
-        with:
-          context: .
-          build-args: PYTHON_VERSION=${{ matrix.python-version }}
-          tags: ${{ steps.metadata.outputs.tags }}
-          labels: ${{ steps.metadata.outputs.labels }}
-          outputs: type=docker,dest=/tmp/image.tar
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
-
-      - name: Test docker image
-        run: |
-          docker load --input /tmp/image.tar
-          docker run --rm prefecthq/prefect:sha-${GITHUB_SHA::7}-python${{ matrix.python-version }} prefect version
-
-      - name: Upload image artifact
-        uses: actions/upload-artifact@v3
-        # Images are stashed for publish after approval is given
-        with:
-          name: release-image-${{ github.sha }}-${{ matrix.python-version }}
-          path: /tmp/image.tar
-
-  build-conda-docker-images:
-    name: Build Conda Docker images
-    runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version:
-          - "3.7"
-          - "3.8"
-          - "3.9"
-          - "3.10"
-          # Not yet supported, see note at top
-          # - "3.11"
-
-    steps:
-      # Not needed until we want multiarchecture builds
-      # - name: Set up QEMU
-      #   uses: docker/setup-qemu-action@v1
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
+      - name: Export CURRENT_BRANCH env var
+        # can't use github.ref_name because that will be e.g. '2.6.9' on tag push, and we want to cache-from main branch on release
+        run: echo "CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD | sed 's/\//-/g')" >> $GITHUB_ENV
 
       - name: Generate image metadata
         id: metadata
         uses: docker/metadata-action@v4
         with:
           images: prefecthq/prefect
```

```suggestion
          images: prefecthq/prefect${{ ( github.event_name == 'release' && '' || '-dev' }}
```

@madkinsz is this what you mean? this would cause a release event to run without cache because `main` tags would only be pushed to `prefect-dev`

---

Yeah I think so — but it'd be nice if it was all in a separate workflow for now so we can keep using our current release workflow while we iterate on this work.

---

amend: like this the `main` tags for `prefecthq/prefect` would only be built on releases, so effectively the last release would be used as cache for the new release

---

Whilst image build time is important, I think getting multiarchitecture builds for `main` merged is the first step. The more we can break this work down, the better. For example, this work could be split into:

- Image builds on main
- Image publish on main
- Image build caching for main
- Image builds on release
- Image publish on release
- Image build caching for release
- Consolidation of image build and publish logic
- Manual dispatch of image build and publish

---

as all above points are included in this PR: do I understand correctly you'd like to merge it into master in separate PRs?

like, make a new yaml, start merging into it with cherry pick PRs, and as final PR remove this yaml?